### PR TITLE
modify shardID type from uint64 to common.ShardID

### LINKED ISSUE #1007: modify shardID type from uint64 to common.ShardID
In sharding branch,modify shardID type from uint64 to common.ShardID

### PR DESCRIPTION
**Pull request by xiemylogos**
1、modify shardID type from uint64 to common.ShardID
close #1007

### CODE REVIEW

File: core/store/ledgerstore/ledger_store.go [hunk 1]
```diff
@@ -655,13 +650,14 @@ func (this *LedgerStoreImp) executeBlock(block *types.Block) (result store.Execu
 	// sort shard Txs
 	ids := make([]uint64, 0)
 	for shardId := range block.ShardTxs {
-		ids = append(ids, shardId)
+		ids = append(ids, shardId.ToUint64())
```
Comment (Honglei-Cong) [inline core/store/ledgerstore/ledger_store.go]: change ids to []common.ShardID ?

File: cmd/contract_cmd.go [hunk 2]
```diff
@@ -297,7 +297,7 @@ func invokeContract(ctx *cli.Context) error {
 
 	paramData, _ := json.Marshal(params)
 	PrintInfoMsg("Invoke:%x Params:%s", contractAddr[:], paramData)
-	shardId := ctx.Uint64(utils.GetFlagName(utils.ShardIDFlag))
+	shardId := common.NewShardIDUnchecked(ctx.Uint64(utils.GetFlagName(utils.ShardIDFlag)))
```
Comment (Honglei-Cong) [inline cmd/contract_cmd.go]: This error should not ignored.

File: core/types/block.go [hunk 3]
```diff
@@ -40,14 +40,14 @@ func (b *Block) Serialization(sink *common.ZeroCopySink) {
 	shardIds := make([]uint64, 0, len(b.ShardTxs))
 
 	for id := range b.ShardTxs {
-		shardIds = append(shardIds, id)
+		shardIds = append(shardIds, id.ToUint64())
```
Comment (Honglei-Cong) [inline core/types/block.go]: also change this shardIds from []uint64 to []common.ShardID ?

File: p2pserver/message/types/ping_test.go [hunk 4]
```diff
@@ -26,12 +26,12 @@ import (
...
 func TestPingSerializationDeserialization(t *testing.T) {
 	var msg Ping
-	msg.Height = make(map[uint64]*HeightInfo)
+	msg.Height = make(map[common.ShardID]*HeightInfo)
 	heightInfo := &HeightInfo{
 		Height:  uint32(1),
 		MsgHash: common.Uint256{1, 2, 3},
 	}
-	msg.Height[0] = heightInfo
+	msg.Height[common.NewShardIDUnchecked(0)] = heightInfo
```
Comment (Honglei-Cong) [inline p2pserver/message/types/ping_test.go]: define one const var for "common.NewShardIDUnchecked(0)" with ShardID?

then, replace all "NewShardIDUnchecked(0)" with that const var.

File: core/store/ledgerstore/ledger_store.go (hunk 1 above)
Comment (xiemylogos) [reply core/store/ledgerstore/ledger_store.go]: yes

File: cmd/contract_cmd.go (hunk 2 above)
Comment (xiemylogos) [reply cmd/contract_cmd.go]: ok,use  NewShardID() get shardID

File: core/types/block.go (hunk 3 above)
Comment (xiemylogos) [reply core/types/block.go]: ok

File: p2pserver/message/types/ping_test.go (hunk 4 above)
Comment (xiemylogos) [reply p2pserver/message/types/ping_test.go]: ok. var RootShardID = NewShardIDUnchecked(0) in shardid.go

**Review by Honglei-Cong — APPROVED** [review]

**Pull request merged**